packages/react-sdk/docusaurus/docs/React/06-ui-cookbook/08-permission-requests.mdx

```diff
@@ -3,4 +3,368 @@ title: Permission Requests
 description: Permission Requests
 ---
 
-Example of how to render the permission requests.
+Permission requests allow call participants to request permission to publish audio/video/screen-share. Participants with elevated permissions can accept or reject these requests. This guide shows you how you can build a UI for sending and receiving these requests.
+
+:::note
+
+The [`ToggleVideoPublishingButton`](/TODO), [`ToggleAudioPublishingButton`](/TODO) and [`ScreenShareButton`](TODO) components can send permission requests, you only need to follow this guide if you want to build your own UI for this.
+
+The [`PermissionRequests`](TODO) component can receive permission requests, you only need to follow this guide if you want to build your own UI for this.
+
+:::
+
+## Prerequisites
+
+Before we can join a call, we need to connect to Stream's edge infrastructure.
+To do that, we follow these steps:
+
+- [Register for a Stream account](https://getstream.io/try-for-free/) and obtain our API key and secret.
+- Install the Stream React Video SDK:
+  - `npm install @stream-io/video-react-sdk`
+  - `yarn add @stream-io/video-react-sdk`
+- [Initialize the SDK](../../guides/client-auth) by passing in your API key, token and user information:
+
+:::note
+For the token generation, you can use our [Token Generator](https://getstream.io/chat/docs/token_generator/).
+:::
+
+## Start point
+
+Let's start with a simple call UI where we create an `audio_room` call.
+
+```tsx
+import {
+  useCreateStreamVideoClient,
+  StreamVideo,
+  StreamCall,
+  CallParticipantsView,
+  CallControls,
+  useCall,
+  StreamTheme,
+} from '@stream-io/video-react-sdk';
+import '@stream-io/video-react-sdk/dist/css/styles.css';
+
+export default function App() {
+  const client = useCreateStreamVideoClient({
+    apiKey: '<API Key>',
+    tokenOrProvider: '<token provider>',
+    user: {
+      id: '<user id>',
+    },
+  });
+
+  return (
+    <StreamTheme>
+      <StreamVideo client={client}>
+        <StreamCall
+          callId={'test-call-permission-requests'}
+          callType={'audio_room'}
+          data={{ create: true }}
+          autoJoin
+        >
+          <CallUI />
+        </StreamCall>
+      </StreamVideo>
+    </StreamTheme>
+  );
+}
+
+const CallUI = () => {
+  const call = useCall();
+
+  return (
+    <div>
+      <PermissionRequestNotifications />
+      <CallParticipantsView call={call} />
+      <CallControls />
+      <PermissionRequests />
+    </div>
+  );
+};
+
+const PermissionRequests = () => {};
+
+const PermissionRequestNotifications = () => {};
+```
+
+:::note
+
+Permission requests are only relevant for [call types](../../guides/10-configuring-call-types/) that have roles where a participant can't publish their audio/video/screen-share by default.
```

Comment:
Would it be contra productive to enumerate those call types in this note?

Reply:
Reading the referenced doc, it was not immediately clear to me where is the dividing line in this aspect among the call types.

Reply:
To be honest I don't know what those call types are, but yeah, we should include that information in the docs one we know them